src/AuthContext.tsx

```diff
@@ -109,16 +109,17 @@ export const AuthProvider = ({ authConfig, children }: IAuthProvider) => {
     }
   }
 
-  function refreshAccessToken(initial = false) {
+  function refreshAccessToken(initial = false): void {
     if (token && epochTimeIsPast(tokenExpire)) {
       if (refreshToken && !epochTimeIsPast(refreshTokenExpire)) {
         fetchWithRefreshToken({ config, refreshToken })
           .then((result: TTokenResponse) => handleTokenResponse(result))
-          .catch((error: string) => {
-            setError(error)
-            if (errorMessageForExpiredRefreshToken(error)) {
-              logOut()
-              redirectToLogin(config)
+          .catch((error: Error) => {
+            console.error(error)
+            setError(error.message)
+            if (initial) login() // If the attempt to get a new token failed during page load, do a full login.
```

Comment:
Did `login` clear the state? Or was that not really needed?

Reply:
Not needed. All relevant state is overwritten with the login anyhow